content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,153 @@
+
+- [Node Selectors](/docs/concepts/scheduling-eviction/assign-pod-node/) are
+  key-value pairs as part of the pod specification that specifies which nodes
+  to deploy onto. These can be enforced at the namespace and cluster level with
+  the
+  [PodNodeSelector](/docs/reference/access-authn-authz/admission-controllers/#podnodeselector)
+  admission controller.
+- [PodTolerationRestriction](/docs/reference/access-authn-authz/admission-controllers/#podtolerationrestriction)
+  is an admission controller that allows administrators to add permitted
+  [tolerations](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+  within a namespace. Pods within that namespace may only utilise those
+  specified in the namespace.
+
+## Secrets
+
+Secrets required for pods should be stored within Kubernetes Secrets as opposed
+to alternatives such as ConfigMap. Secret resources stored within etcd should
+be [encrypted at rest](/docs/tasks/administer-cluster/encrypt-data/).
+
+Pods needing secrets should have these automatically mounted through volumes,
+preferably stored in memory like with the
+[`emptyDir.medium` option](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir).
+This should be done preferentially as compared to providing the pods service
+account RBAC access to secrets. This would allow adding secrets into the pod as
+environment variables or files. However, the environment variable method might
+be more prone to leakage due to crash dumps in logs and the non-confidential
+nature of environment variable in Linux, as opposed to the permission mechanism
+on files.
+
+Service accounts should not be mounted into pods that do not require them. This can be configured by setting
+[`automountServiceAccountToken`](/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)
+to `false` either within the service account to apply throughout the namespace
+or specifically for a pod. For Kubernetes v1.22 and above, use
+[Bound Service Accounts](/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+for time-bound service account credentials.
+
+## Container & Image scanning
+
+Scanning a container image can prevent critical vulnerabilities from being
+deployed to the cluster alongside the container image. Image scanning should be
+completed before deploying a container image to a cluster and is usually done
+as part of the deployment process in a CI/CD pipeline. The purpose of an image
+scan is to obtain information about possible vulnerabilities and their
+prevention in the container image, such as a
+[Common Vulnerability Scoring System (CVSS)](https://en.wikipedia.org/wiki/Common_Vulnerability_Scoring_System)
+score. If the result of the image scans is combined with the pipeline
+compliance rules, only properly patched container images will end up in
+Production.
+
```

Comment:
Might it be good to point towards CIS benchmark as well.

https://www.cisecurity.org/benchmark/kubernetes

Also pointing to kube-bench or similar tools.
https://github.com/aquasecurity/kube-bench

Reply:
Our [actual guide](https://kubernetes.io/docs/contribute/style/content-guide/#what-s-allowed) to what's allowed.

If the guide isn't clear, PRs (or issues) against the guide are welcome!

Reply:
@sftim it's very clear. The question is if it's good :)
But that is for another PR i guess <3

Reply:
Regarding the guide, is there anything that you still want to add @NissesSenap?

Reply:
I will resolve this thread since it's almost three months old!